### ISSUE TITLE
Slider to affect sharpness would be a great addition

### ISSUE THREAD

**Issue by 7Brandyn7** [Blinue/Magpie#61]
Right below the options would be a great one.

It just has to change the sharpness to a 0-10 slider then divide by 10

![Mockup](https://user-images.githubusercontent.com/25763206/129407093-81ce5eb7-8c8d-485a-b647-80595a965f29.PNG)

```
        "module": "FFX",
        "effect": "FsrRcas",
        "sharpness": 0.4
```

**Comment by Blinue** [comment]
While this is a convenient option, I would like to decouple the UI from the specific scaling effect. Users may delete, modify, or rename FSR effect, and I'm not sure when this option should be available.

If you need this feature, please implement it in your own branch.

**Comment by 7Brandyn7** [comment]
@Blinue 
What if you just do it for sharpening in general.

**Comment by Blinue** [comment]
There are countless combinations of zoom effects, for example: 
```
[
    {
        "module": "FFX",
        "effect": "FsrEasu",
        "scale": [ -1, -1 ]
    },
    {
        "module": "FFX",
        "effect": "FsrRcas",
        "sharpness": 0.5
    },
    {
        "module": "FFX",
        "effect": "CAS",
        "sharpness": 0.5
    }
]
```
There are two sharpening effects here, which one should be adjusted by this option?

**Comment by 7Brandyn7** [comment]
I would say it should do both.

**Comment by Blinue** [comment]
This makes too many assumptions. I don't think you can adjust ScaleModels with a simple parameter. The best solution is to create a UI for the whole ScaleModels, but for now you can only adjust it by editing ScaleModels.json.

**Comment by soredake** [comment]
> The best solution is to create a UI for the whole ScaleModels

It would be super nice to have this. I've tested some game with this:
```
	    "module": "FFX",
        "effect": "CAS",
        "sharpness": 0.3
      },
      {
	    "module": "FFX",
        "effect": "FSR",
        "sharpness": 0.3
      }
```
and without this, game looks better without this in the end.

**Comment by Blinue** [comment]
You can try as much as you like to find the one that suits you best

**Comment by MuhammadUsmanMithani** [comment]
can u guys tell me what is the difference between cas and fsr sharpness effect, i couldnt understand
I watched @7Brandyn7  video but i didnt understand the difference between both
which is executed when u upscale by hotkey?
I made a video on my channel, mythtech abt magpie but saw brandyns video after it
I need some info on cas vs fsr please, since both go under ffx module
thanks in advance

**Comment by Blinue** [comment]
FSR and CAS are completely different. CAS is a sharpening algorithm, and FSR is a scaling algorithm. FSR itself contains a sharpening step. If you think that FSR is not sharp enough, you can add another sharpening step afterwards, such as CAS.

**Comment by soredake** [comment]
Would be nice to have FSR present too like in this program https://store.steampowered.com/news/app/993090/view/2957161686744542612

**Comment by Blinue** [comment]
Implemented in v0.10 🎉
